### PR TITLE
Generate contentMetadata.xml for the uploaded files

### LINKED ISSUE #54: Create contentMetadata.xml
See https://github.com/sul-dlss-deprecated/dor-scripts/pull/31/files as an example.

### PR DESCRIPTION
**Pull request by jcoyne**
## Why was this change made?

So that the assembly WF has the required prerequisites.
Fixes #54

## Was the documentation (README.md, openapi.yml) updated?

n/a

### CODE REVIEW

File: app/jobs/ingest_job.rb [hunk 1]
```diff
@@ -4,11 +4,20 @@
...
+  # @param [Array<Hash>] filesets the data for creating the structure
+  # @param [BackgroundJobResult] background_job_result
+  def perform(druid:, filesets:, background_job_result:)
     background_job_result.processing!
 
-    # TODO: generate contentMetadata.xml
-    # TODO: ship files
+    dir = StagingDirectory.new(druid: druid, staging_location: Settings.staging_location)
+
+    file_names = copy_files_to_staging(dir, filesets)
```
Comment (justinlittman) [inline app/jobs/ingest_job.rb]: Should this copy be verified? (That is, verify checksums).

File: app/services/content_metadata_generator.rb [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+  def cm_params
+    {
+      druid: druid,
+      objects: object_files,
+      file_attributes: file_attributes,
+      add_file_attributes: true,
+      add_exif: false,
+      bundle: :filename,
+      style: :simple_book
```
Comment (justinlittman) [inline app/services/content_metadata_generator.rb]: If this is a temporary hardcode, can you please call out with a comment?

File: app/controllers/resources_controller.rb [hunk 3]
```diff
@@ -15,7 +15,10 @@ def create
     end
 
     result = BackgroundJobResult.create
-    IngestJob.perform_later(druid: response[:pid], background_job_result: result)
+
+    IngestJob.perform_later(druid: response[:pid],
```
Comment (justinlittman) [inline app/controllers/resources_controller.rb]: I don't see a test for the IngestJob invocation.

File: app/jobs/ingest_job.rb (hunk 1 above)
Comment (jcoyne) [reply app/jobs/ingest_job.rb]: I don't think this is something we do when we do preassembly, so no.  If you feel strongly about it, open a ticket and we can discuss it with the PO.

File: app/controllers/resources_controller.rb (hunk 3 above)
Comment (jcoyne) [reply app/controllers/resources_controller.rb]: Okay, added one.

File: app/services/content_metadata_generator.rb (hunk 2 above)
Comment (jcoyne) [reply app/services/content_metadata_generator.rb]: I don't know the answer to that without feedback from Andrew.

File: app/controllers/resources_controller.rb (hunk 3 above)
Comment (mjgiarlo) [reply app/controllers/resources_controller.rb]: I don't see one in the branch yet FWIW!

File: app/jobs/ingest_job.rb (hunk 1 above)
Comment (justinlittman) [reply app/jobs/ingest_job.rb]: https://github.com/sul-dlss/google-books/issues/110

**Comment by jcoyne** [comment]
@mjgiarlo the code in StagingDirectory and ContentMetadataGenerator is well covered by the tests for IngestJob.  These classes support IngestJob in doing its work.  So the tests for IngestJob acts as an integration test for all of them.  I'm trying to follow the advice of Ken C. Dodds - "Write tests. Not too many. Mostly integration."

File: app/controllers/resources_controller.rb (hunk 3 above)
Comment (jcoyne) [reply app/controllers/resources_controller.rb]: oops, forgot to push.

**Pull request merged**